docs/using_python_client_with_hazelcast_imdg.rst

```diff
@@ -1321,6 +1321,304 @@ the entry processor in the ``Map`` methods. See the following example.
+
+- For :ref:`Portable<serialization:Portable Serialization>` objects, the fields
+  that are written in the
+  :func:`write_portable<hazelcast.serialization.api.Portable.write_portable>`
+  method are exposed using their exact names.
+- For
+  :ref:`IdentifiedDataSerializable<serialization:IdentifiedDataSerializable Serialization>`
+  objects, the object is deserialized if needed and then analyzed using the
+  reflection mechanism (on the server-side). Only public fields and getters
+  are taken into account. See the `IMDG Reference Manual
+  <https://docs.hazelcast.com/imdg/latest/sql/querying-imap.html#key-and-value-fields>`__
+  for details.
+
+.. note::
+
+    You cannot query JSON fields in SQL. If you want to query JSON, see
+    :ref:`using_python_client_with_hazelcast_imdg:Querying with JSON Strings`
+    section.
+
+
+Consider the ``Employee`` class from the example above; the SQL service can
+access the following fields:
+
+==== =======
+Name Type
+==== =======
+name VARCHAR
+age  INTEGER
+==== =======
+
+Together with the key and value objects, you may query the following fields
+from the map:
+
+.. code:: sql
+
+    SELECT __key, this, name, age FROM employee
+
+If both the key and value have fields with the same name, then the field of
+the value is exposed.
+
+**"SELECT *" Queries**
+
+You may use the ``SELECT * FROM <table>`` syntax to get all the table fields.
+
+The ``__key`` and ``this`` fields are returned by the ``SELECT *`` queries if
+they do not have nested fields. For the ``employee`` map, the following query
+does not return the ``this`` field, because the value has nested fields
+``name`` and ``age``:
+
+.. code:: sql
+
+    -- Returns __key, name, age
+    SELECT * FROM employee
+
+**Indexes**
+
+The SQL service can use ``Map`` indexes to speed up the execution of certain
+queries. ``SORTED`` and ``HASH`` indexes are supported.
+
+Data Types
+~~~~~~~~~~
+
+The SQL service supports a set of SQL data types. Every data type is mapped to
+a Python type that represents the type’s value.
+
+======================== ===============
+Type Name                Python Type
+======================== ===============
+BOOLEAN                  bool
+VARCHAR                  str
+TINYINT                  int
+SMALLINT                 int
+INTEGER                  int
+BIGINT                   int
+DECIMAL                  str
+REAL                     float
+DOUBLE                   float
+DATE                     str
+TIME                     str
+TIMESTAMP                str
+TIMESTAMP_WITH_TIME_ZONE str
+OBJECT                   Any Python type
+======================== ===============
+
+Note that, the following types are returned as strings, with the following
+formats.
+
+- ``DATE`` with the ``YYYY-MM-DD`` format.
+- ``TIME`` with the ``HH:MM:SS[.mmmmmm]`` format.
+- ``TIMESTAMP`` with the ``YYYY-MM-DDTHH:MM:SS[.ffffff]`` format.
+- ``TIMESTAMP_WITH_TIME_ZONE`` with the ``YYYY-MM-DDTHH:MM:SS.ffffff+HH:MM[:SS[.ffffff]]``
```

Comment:
I don't think ffff after in [:SS[.ffffff]] is possible here since offset is in seconds